src/goto-checker/multi_path_symex_checker.cpp

```diff
@@ -0,0 +1,238 @@
+/*******************************************************************\
+
+Module: Goto Checker using Bounded Model Checking
+
+Author: Daniel Kroening, Peter Schrammel
+
+\*******************************************************************/
+
+/// \file
+/// Goto Checker using Bounded Model Checking
+
+#include "multi_path_symex_checker.h"
+
+#include <chrono>
+
+#include "bmc_util.h"
+#include "counterexample_beautification.h"
+
+multi_path_symex_checkert::multi_path_symex_checkert(
+  const optionst &options,
+  ui_message_handlert &ui_message_handler,
+  abstract_goto_modelt &goto_model)
+  : multi_path_symex_only_checkert(options, ui_message_handler, goto_model),
+    symex_has_run(false)
+{
+  solver_factoryt solvers(
+    options,
+    ns,
+    ui_message_handler,
+    ui_message_handler.get_ui() == ui_message_handlert::uit::XML_UI);
+  solver = solvers.get_solver();
+  solver->prop_conv().set_message_handler(ui_message_handler);
+}
+
+incremental_goto_checkert::resultt multi_path_symex_checkert::
+operator()(propertiest &properties)
+{
+  resultt result;
+
+  if(!symex_has_run)
+  {
+    perform_symex();
+    output_coverage_report();
+    update_properties_status_from_symex_target_equation(
+      properties, result.updated_properties, equation);
+    // Since we will not symex any further we can decide the status
+    // of all properties that do not occur in the equation now.
+    // The current behavior is PASS.
+    update_properties_status_not_checked(properties, result.updated_properties);
+  }
+
+  // have we got anything to check? otherwise we return DONE
+  if(!has_properties_to_check(properties))
+    return result;
+
+  prop_convt &prop_conv = solver->prop_conv();
+
+  auto solver_start = std::chrono::steady_clock::now();
+
+  if(!symex_has_run)
```

Comment:
Name no longer accurate, since symex definitely has run now.

Reply:
Renamed to `equation_generated`